common/presenters/table/object-to-table-head.js

```diff
@@ -1,17 +1,61 @@
 const { pickBy } = require('lodash')
+const queryString = require('query-string')
 
 const i18n = require('../../../config/i18n')
 
-function objectToTableHead({ head } = {}) {
-  if (!head) {
-    return undefined
+function buildUrl({ key }, query = {}) {
+  return queryString.stringify({
+    ...query,
+    sortBy: key,
+    sortDirection: query.sortDirection === 'asc' ? 'desc' : 'asc',
+  })
+}
+
+function buildAriaLabel(head, query) {
+  // the label indicates the result of the sorting action triggered by the link,
+  // which is the opposite of the current state
+  return `${i18n.t('sort')} ${i18n.t(head.html).toLowerCase()} ${
+    query.sortDirection === 'asc' ? i18n.t('descending') : i18n.t('ascending')
+  }`
+}
+
+function buildWrapper(head, query) {
+  const label = i18n.t(head.html)
+  if (!head.isSortable) {
+    return label
+  }
+  const url = buildUrl(head, query)
+  const ariaLabel = buildAriaLabel(head, query)
+  return `<a aria-label="${ariaLabel}" class="sortable-table__button" role="button" href="?${url}">${label}</a>`
+}
+
+function assignSortAttributes(head, query) {
+  if (!head.isSortable) {
+    return head.attributes ? { ...head.attributes } : undefined
+  }
+  const isCurrentSortColumn = head.key === query.sortBy
+  return {
+    ...head.attributes,
+    'aria-sort': isCurrentSortColumn
+      ? query.sortDirection === 'asc'
+        ? 'ascending'
+        : 'descending'
+      : 'none',
   }
```

Comment:
I do find nested ternaries a bit head-melting. Could we not do something like

```js
  let ariaSort = 'none'
  if (head.key === query.sortBy) {
    const ariaSortLabels = {
      asc: 'ascending',
      desc: 'descending'
    }
    ariaSort = ariaSortLabels[query.sortDirection]
  }
  return {
    ...head.attributes,
    'aria-sort': ariaSort
  }
```

Reply:
actually, having hoisted that out of the return statement, there's probably little to be gained from the object map now...

Reply:
I have adopted @teneightfive suggestion (see below) of using an intermediate variable. Does it look better now?